### PR TITLE
Dump SDK API error

### PR DESCRIPTION
**Pull request by TaoZou1**
Dump detail of SDK API return error

Test Done:
1. change PageSize=2000 to trigger error
2. check if message is dumped

2024-05-10 07:28:22.635 INFO    common/store.go:105     translate error {"apierror": {"Details":"Field level validation errors: {value 1,100 of property page_size has exceeded its maximum valid value 1000}","ErrorCode":255,"ErrorData":null,"ErrorMessage":"Field level validation errors: {value 1,100 of property page_size has exceeded its maximum valid value 1000}","ModuleName":"common-services","RelatedErrors":null}, "error type": "INVALID_REQUEST"}

### CODE REVIEW

File: pkg/controllers/staticroute/staticroute_controller.go [hunk 1]
```diff
@@ -97,6 +98,10 @@ func (r *StaticRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if err := r.Service.CreateOrUpdateStaticRoute(req.Namespace, obj); err != nil {
 			updateFail(r, &ctx, obj, &err)
 			// TODO: if error is not retriable, not requeue
+			apierror, _ := util.DumpAPIError(err)
```
Comment (dantingl) [inline pkg/controllers/staticroute/staticroute_controller.go:101]: Is this test code?

File: pkg/controllers/staticroute/staticroute_controller.go [hunk 2]
```diff
@@ -97,6 +98,10 @@ func (r *StaticRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		if err := r.Service.CreateOrUpdateStaticRoute(req.Namespace, obj); err != nil {
 			updateFail(r, &ctx, obj, &err)
 			// TODO: if error is not retriable, not requeue
+			apierror, errortype := util.DumpAPIError(err)
```
Comment (zhengxiexie) [inline pkg/controllers/staticroute/staticroute_controller.go:101]: A nit suggestion, if apierror is not nil, could we write it to status, so we can debug from CR status more straight forward?

**Comment by zhengxiexie** [comment]
squash into 1 commit.

**Comment by TaoZou1** [comment]
There is two commits here, close it and create a new one

**Pull request closed without merging**